lib/re1.5/compilecode.c

```diff
@@ -18,19 +18,21 @@ static void _emit_checked(int at, char *code, int val, bool *err) {
     }
 }
 
-static const char *_compilecode(const char *re, ByteProg *prog, int sizecode)
+static const char *_compilecode(const char *re, size_t *len, ByteProg *prog, int sizecode)
```

Comment:
Does this function need to pass the updated `len` back to the caller?  If not it's more efficient to just have `size_t len`.

Reply:
I did need to pass the len back to track how much is consumed when the function is run recursively, but this need goes away with a change to `re_top` :-)